sdk/go/common/workspace/plugins.go

```diff
@@ -1660,6 +1660,21 @@ func getPlugins(dir string, skipMetadata bool) ([]PluginInfo, error) {
 	return plugins, nil
 }
 
+// We currently bundle some plugins with "pulumi" and thus expect them to be next to the pulumi binary.
+// Eventually we want to fix this so new plugins are true plugins in the plugin cache.
+func IsPluginBundled(kind PluginKind, name string) bool {
+	return (kind == LanguagePlugin && name == "nodejs") ||
+		(kind == LanguagePlugin && name == "go") ||
+		(kind == LanguagePlugin && name == "python") ||
+		(kind == LanguagePlugin && name == "dotnet") ||
+		(kind == LanguagePlugin && name == "yaml") ||
+		(kind == LanguagePlugin && name == "java") ||
+		(kind == ResourcePlugin && name == "pulumi-nodejs") ||
+		(kind == ResourcePlugin && name == "pulumi-python") ||
+		(kind == AnalyzerPlugin && name == "policy") ||
+		(kind == AnalyzerPlugin && name == "policy-python")
```

Comment:
It's too bad we don't have pattern matching.

This is completely fine as-is, but if you'd prefer for this to not be a series of `&&`/`||`s, you can use a map/set with a composite key.

```go
type bundledPlugin struct{ kind PluginKind; name string }
```

You can either declare a map, e.g.,

```go
var bundledPlugins = map[bundledPlugin]struct{}{
  {LanguagePlugin, "nodejs"}: {},
  {LanguagePlugin, "go"}: {},
  // ..
}

func IsPluginBundled(kind PluginKind, name string) bool {
	_, ok := bundledPlugins[bundledPlugin{kind, name}]
	return ok
}
```

Or if you'd prefer to not add the `: {}`, you can declare a slice and compile the map at init.

```go
var bundledPlugins = []bundledPlugin{
	{LanguagePlugin, "nodejs"},
	{LanguagePlugin, "go"},
	// ...
}

// TODO: Do we have a shared set type we can use here
// and do something like `set.FromSlice[bundledPlugin](bundledPlugins)`?
var bundledPluginSet = make(map[bundledPlugin]struct{})

func init() {
	for _, p := range bundledPlugins {
		bundledPluginSet[p] = struct{}{}
	}
}
```

Reply:
It's simple to read, it's only in one place, we'll shrink this over time hopefully, probably fine as is.